bin/deployables

```diff
@@ -83,600 +99,829 @@ region=${DEPLOY_AWS_REGION}
+      )
+
+    if [ -z "$revision_deployed" ] ; then
+      echo "Success: Deployed!"
+      return 0
     fi
 
-    # part one of some bash craziness to pass the revision_target
-    key="revision_target_$( echo $family | tr -s - _ )"
-    eval "$key=${revision_target}"
-    echo key : $key : ${!key}
+    echo "Waiting for update:"
+    echo "  revision_target: ${revision_target}"
+    echo "  revision_deployed: ${revision_deployed}"
+    sleep 6
+
+  done
 
+  echo "Error: Service update took too long."
+  exit 1
 }
 
-function ecs_verify_deploy () {
 
-    local subfamily="$1"
+##
+## public functions
+##
 
-    if [ -z "$subfamily" ] ; then
-        family="${DEPLOY_APP_NAME}"
-        service="${DEPLOY_APP_NAME}"
-    else
-        family="${DEPLOY_APP_NAME}-${subfamily}"
-        service="${DEPLOY_APP_NAME}-${subfamily}"
-    fi
+function docker_build {
 
-    # part two of some bash craziness to receive the revision_target
-    key="revision_target_$( echo $family | tr -s - _ )"
-    revision_target=${!key}
+  echo
+  echo Building docker image
+  echo DEPLOY_DOCKERFILE: ${DEPLOY_DOCKERFILE}
+  echo DEPLOY_DOCKER_LOCAL_TAG: ${DEPLOY_DOCKER_LOCAL_TAG}
+  echo
 
-    echo
-    echo Verifying deploy
-    echo revision_target: $revision_target
-    echo
+  if [ -z "${DEPLOY_DOCKERFILE}" ] ; then
+    echo "Error: DEPLOY_DOCKERFILE is required"
+    return 1
+  fi
 
-    # wait for older revisions to disappear
-    for attempt in {1..60}; do
+  if [ -z "${DEPLOY_DOCKER_LOCAL_TAG}" ] ; then
+    echo "Error: DEPLOY_DOCKER_LOCAL_TAG is required"
+    return 1
+  fi
 
-        revision_deployed=$( \
-            aws ecs describe-services \
-                --profile ${DEPLOY_AWS_PROFILE} \
-                --cluster ${DEPLOY_ECS_CLUSTER_NAME} \
-                --services ${service} \
-                | $JQ ".services[0].deployments | .[] | select(.taskDefinition != \"$revision_target\") | .taskDefinition" \
-            )
+  if [[ ! -z "${DEPLOY_GITHUB_MACHINE_USER_KEY_FINGERPRINT}" ]] ; then
 
-        if [ -z "$revision_deployed" ] ; then
-            echo "Success: Deployed!"
-            return 0
-        fi
+    fingerprint="$( echo ${DEPLOY_GITHUB_MACHINE_USER_KEY_FINGERPRINT} | tr -d : )"
 
-        echo "Waiting for update:"
-        echo "  revision_target: ${revision_target}"
-        echo "  revision_deployed: ${revision_deployed}"
-        sleep 5
+    eval keyfile="~/.ssh/id_${fingerprint}"
 
-    done
+    if [[ ! -f ${keyfile} ]] ; then
+      echo "Error: Unable to find machine user key file"
+      echo "- fingerprint: ${fingerprint}"
+      echo "- keyfile: ${keyfile}"
+
+      eval keyfile="~/.ssh/id_circleci_github"
+
+      if [[ ! -f ${keyfile} ]] ; then
+        echo "Error: Unable to find circle github key file"
+        echo "- keyfile: ${keyfile}"
+        return 1
+      fi
+
+    fi
+
+    echo "Using GITHUB_MACHINE_USER_KEY=${keyfile}"
+
+    docker build \
+      --rm=false \
+      --no-cache \
```

Comment:
Is there a reason why this `docker build` call passes in the `--no-cache` flag, but the other `docker build` (just below) does not?

Reply:
Oh, whoa. That's not good.